core/src/main/java/org/apache/iceberg/PartitionStatsUtil.java

```diff
@@ -0,0 +1,213 @@
+    checkAndIncrementLong(toUpdate, fromRecord, Column.EQUALITY_DELETE_RECORD_COUNT);
+    checkAndIncrementInt(toUpdate, fromRecord, Column.EQUALITY_DELETE_FILE_COUNT);
+    checkAndIncrementLong(toUpdate, fromRecord, Column.TOTAL_RECORD_COUNT);
+    if (toUpdate.get(Column.LAST_UPDATED_AT.ordinal()) != null
+        && fromRecord.get(Column.LAST_UPDATED_AT.ordinal()) != null
+        && ((long) toUpdate.get(Column.LAST_UPDATED_AT.ordinal())
+            < (long) fromRecord.get(Column.LAST_UPDATED_AT.ordinal()))) {
+      toUpdate.set(
+          Column.LAST_UPDATED_AT.ordinal(), fromRecord.get(Column.LAST_UPDATED_AT.ordinal()));
+      toUpdate.set(
+          Column.LAST_UPDATED_SNAPSHOT_ID.ordinal(),
+          fromRecord.get(Column.LAST_UPDATED_SNAPSHOT_ID.ordinal()));
+    }
+  }
+
+  public static Record partitionDataToRecord(
+      Types.StructType partitionSchema, PartitionData partitionData) {
+    GenericRecord genericRecord = GenericRecord.create(partitionSchema);
+    for (int index = 0; index < partitionData.size(); index++) {
+      genericRecord.set(index, partitionData.get(index));
+    }
+
+    return genericRecord;
+  }
+
+  private static Record fromManifestEntry(
+      ManifestEntry<?> entry, Table table, Schema recordSchema) {
+    GenericRecord record = GenericRecord.create(recordSchema);
+    Types.StructType partitionType =
+        recordSchema.findField(Column.PARTITION_DATA.name()).type().asStructType();
+    PartitionData partitionData = coercedPartitionData(entry.file(), table.specs(), partitionType);
+    record.set(
+        Column.PARTITION_DATA.ordinal(), partitionDataToRecord(partitionType, partitionData));
+    record.set(Column.SPEC_ID.ordinal(), entry.file().specId());
+
+    Snapshot snapshot = table.snapshot(entry.snapshotId());
+    if (snapshot != null) {
+      record.set(Column.LAST_UPDATED_SNAPSHOT_ID.ordinal(), snapshot.snapshotId());
+      record.set(Column.LAST_UPDATED_AT.ordinal(), snapshot.timestampMillis());
+    }
+
+    switch (entry.file().content()) {
+      case DATA:
+        record.set(Column.DATA_FILE_COUNT.ordinal(), 1);
+        record.set(Column.DATA_RECORD_COUNT.ordinal(), entry.file().recordCount());
+        record.set(Column.DATA_FILE_SIZE_IN_BYTES.ordinal(), entry.file().fileSizeInBytes());
+        break;
+      case POSITION_DELETES:
+        record.set(Column.POSITION_DELETE_FILE_COUNT.ordinal(), 1);
+        record.set(Column.POSITION_DELETE_RECORD_COUNT.ordinal(), entry.file().recordCount());
+        break;
+      case EQUALITY_DELETES:
+        record.set(Column.EQUALITY_DELETE_FILE_COUNT.ordinal(), 1);
+        record.set(Column.EQUALITY_DELETE_RECORD_COUNT.ordinal(), entry.file().recordCount());
+        break;
+      default:
+        throw new UnsupportedOperationException(
+            "Unsupported file content type: " + entry.file().content());
+    }
+
+    // TODO: optionally compute TOTAL_RECORD_COUNT based on the flag
```

Comment:
the idea of this field is to apply the equality or position delete filters and give the effective results. Need to add a logic to compute it in the follow up.

Reply:
That would be tricky, not sure we can actually do it without actually scanning rows.

Reply:
True. But we decided we need to have this optional field during spec design. 

Plan is to have a follow up PR with a flag that enables computing this. By default it won't be computed as it is an expensive operation.

Reply:
I am not sure we would want to extend this path to compute deletes. We may also reconsider how position deletes are handled in V3, so it may be possible to compute this without scanning the data.

I'd probably drop this TODO for now.

Reply:
True. Maybe we didn't think well about this field during spec design. We can drop a todo and make this field deprecated if required later on.